website/addons/base/views.py

```diff
@@ -217,3 +222,73 @@ def get_waterbutler_render_url(**kwargs):
         raise HTTPError(httplib.BAD_REQUEST)
 
     return redirect(url)
+
+
+def get_or_start_render(file_guid, start_render=True, **extras):
+
+    try:
+        if not file_guid.enriched:
+            file_guid.enrich()
+
+        return codecs.open(file_guid.cache_path, 'r', 'utf-8').read()
+    except exceptions.AddonEnrichmentError as error:
+        return error.renderable_error
+    except IOError:
+        if start_render:
+            # Start rendering job if requested
+            download_url = furl.furl(file_guid.download_url)
+            download_url.args.update(extras)
```

Comment:
I'm assuming the **extras contains any addon specific request information such as specific versions, correct?